### PR TITLE
Fix: Nested Tabs styling issues on tabs section [ED-8841]

### PR DESCRIPTION
**Pull request by davseve**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [x] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 1]
```diff
@@ -425,6 +425,9 @@ protected function register_controls() {
 				'fields_options' => [
 					'color' => [
```
Comment (hein-obox) [inline modules/nested-tabs/widgets/nested-tabs.php:426]: Please make sure to update the background gradient selector as well.

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 2]
```diff
@@ -34,7 +34,8 @@
 	--n-tabs-title-background-color: #{$editor-background-light};
 	--n-tabs-title-color-active: #{$white};
 	--n-tabs-title-background-color-active: var( --e-global-color-accent, var( --n-tabs-color-accent-fallback ) );
-	--n-tabs-title-width: initial;
+	--n-tabs-title-background-color-hover: var( --e-global-color-accent, var( --n-tabs-color-accent-fallback ) );
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:37]: Safe to delete?

Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:37]: I will delete them once I'll see that all the backgrounds work well.
Good comment :)

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 3]
```diff
@@ -423,9 +423,6 @@ protected function register_controls() {
 				'exclude' => [ 'image' ],
 				'selector' => "{$nested_tabs_heading_selector_class} > .e-n-tab-title",
 				'fields_options' => [
-					'background' => [
-						'default' => 'classic',
-					],
```
Comment (hein-obox) [inline modules/nested-tabs/widgets/nested-tabs.php]: Why is this removed?

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 4]
```diff
@@ -85,7 +85,7 @@ protected function register_controls() {
 		$end = is_rtl() ? 'left' : 'right';
 		$tooltip_start = is_rtl() ? esc_html__( 'Right', 'elementor' ) : esc_html__( 'Left', 'elementor' );
 		$tooltip_end = is_rtl() ? esc_html__( 'Left', 'elementor' ) : esc_html__( 'Right', 'elementor' );
-		$nested_tabs_heading_selector_class = '{{WRAPPER}} > .elementor-widget-container > .e-n-tabs > .e-n-tabs-heading';
+		$nested_tabs_heading_selector_class = ':is( {{WRAPPER}} > .elementor-widget-container > .e-n-tabs > .e-n-tabs-heading, {{WRAPPER}} > .elementor-widget-container > .e-n-tabs > .e-n-tabs-content )';
```
Comment (hein-obox) [inline modules/nested-tabs/widgets/nested-tabs.php:88]: Could you add a `>` charachter after the ` .e-n-tabs-content`? If we do that, we don't have to add them at the selectors below.

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 5]
```diff
@@ -421,10 +421,13 @@ protected function register_controls() {
 				'name' => 'tabs_title_background_color',
 				'types' => [ 'classic', 'gradient' ],
 				'exclude' => [ 'image' ],
-				'selector' => "{$nested_tabs_heading_selector_class} .e-n-tab-title",
+				'selector' => "{$nested_tabs_heading_selector_class} > .e-n-tab-title",
```
Comment (hein-obox) [inline modules/nested-tabs/widgets/nested-tabs.php:424]: Possibly delete and restore to original version. See comment above.

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 6]
```diff
@@ -972,7 +987,7 @@ protected function register_controls() {
 			Group_Control_Box_Shadow::get_type(),
 			[
 				'name' => 'box_shadow_box_shadow',
-				'selector' => "{$nested_tabs_content_selector_class}",
+				'selector' => '{{WRAPPER}} .e-n-tabs-content',
```
Comment (hein-obox) [inline modules/nested-tabs/widgets/nested-tabs.php:990]: This is incorrect. Now the styling will be applied to child nested tab widgets as well.

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 7]
```diff
@@ -421,10 +421,13 @@ protected function register_controls() {
 				'name' => 'tabs_title_background_color',
 				'types' => [ 'classic', 'gradient' ],
 				'exclude' => [ 'image' ],
-				'selector' => "{$nested_tabs_heading_selector_class} .e-n-tab-title",
+				'selector' => "{$nested_tabs_heading_selector_class} > .e-n-tab-title",
 				'fields_options' => [
 					'color' => [
 						'label' => esc_html__( 'Background Color', 'elementor' ),
+						'selectors' => [
+							'{{WRAPPER}}' => '--n-tabs-title-background-color: {{VALUE}}',
```
Comment (hein-obox) [inline modules/nested-tabs/widgets/nested-tabs.php:429]: I am curious about the logic.

We use a css variable here, but for the hover and active state we use a php selector.

What is the logic?

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 8]
```diff
@@ -229,6 +229,17 @@
 	}
 }
 
+// TODO: Remove once PHP global color support is fixed.
+.elementor .elementor-element.elementor-widget-n-tabs > .elementor-widget-container > .e-n-tabs > .e-n-tabs-heading .e-n-tab-title {
+	&.e-active {
+		background: var( --n-tabs-title-background-color-active );
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:235]: Could it work if we use `background-color` here and then add `background-image: initial;`?

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 9]
```diff
@@ -159,7 +159,7 @@
 			&.e-active {
 				&:not(:hover) {
 					color: var( --n-tabs-title-color-active );
-					background-color: var( --n-tabs-title-background-color-active );
+					background: var( --n-tabs-title-background-color-active );
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:162]: Shouldn't we remove these and move them to the new css selector with the higher specificity?

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 10]
```diff
@@ -191,7 +191,7 @@
 
 			&:hover {
 				color: var( --n-tabs-title-color-hover );
-				background-color: var( --n-tabs-title-background-color-hover );
+				background: var( --n-tabs-title-background-color-hover );
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:194]: Shouldn't we remove these and move them to the new css selector with the higher specificity?

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 11]
```diff
@@ -435,6 +435,86 @@ test.describe( 'Nested Tabs tests', () => {
 
 		await cleanup( wpAdmin );
 	} );
+
+	test( 'Check that active tab receives custom hover color when hovering above it', async ( { page }, testInfo ) => {
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:439]: Suggestion: 'Check if the hover effect works on active tabs`

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 12]
```diff
@@ -435,6 +435,86 @@ test.describe( 'Nested Tabs tests', () => {
...
+
+		await activeTab.hover();
+
+		// Check that active tab receives the hover defined background color.
+		await expect( activeTab ).toHaveCSS( 'background-color', backgroundHoverStyle );
+
+		await cleanup( wpAdmin );
+	} );
+
+	test( 'Check that while hovering an active tab hover styling is shown', async ( { page }, testInfo ) => {
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:475]: Suggestion: 'Check if the hover effect works on active tabs`

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 9 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:162]: I think that the CSS property should be the same in both cases.
In the control, we are targeting the `background` CSS property in the PHP.
If my explanation is not clear I will elaborate on the explanation

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 10 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:194]: I think that the CSS property should be the same in both cases.
In the control, we are targeting the background CSS property in the PHP.
If my explanation is not clear I will elaborate on the explanation.

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 4 above)
Comment (davseve) [reply modules/nested-tabs/widgets/nested-tabs.php:88]: That was my fist try :)
It didn't work, so I added the direct sign to after each and every use of the `nested_tabs_heading_selector_class` variable.

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 7 above)
Comment (davseve) [reply modules/nested-tabs/widgets/nested-tabs.php:429]: The background color should be weaker (less explicit) than the `active` and `hover` states but stronger (more explicit) that the normal background color.
In this way, I achieve exactly that :)

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 6 above)
Comment (davseve) [reply modules/nested-tabs/widgets/nested-tabs.php:990]: Probably missed it in the merging from 3.10.0.
Fixed

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 11 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:439]: Great suggestion.
Done.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js [hunk 13]
```diff
@@ -491,7 +571,9 @@ async function cleanup( wpAdmin ) {
 
 async function setTabItemColor( page, editor, panelClass, tabState, colorPickerClass, color ) {
 	await editor.activatePanelTab( 'style' );
-	await page.locator( `.elementor-control-${ panelClass }` ).click();
+	if ( 'tabs' !== panelClass ) {
```
[the remark this answers is not in the document]
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js:574]: `panelClass` is the class of the panel that I want to open in order to create changes in its controls,
When I choose the `style` tab, the tabs is already open, and therefore, clicking the tabs panel will close it. 
That is why I excluded the tabs panel.
See the image bellow.
image

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 5 above)
Comment (davseve) [reply modules/nested-tabs/widgets/nested-tabs.php:424]: Please see comment above

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 8 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:235]: Changed to your suggestion, 
Looks like it works OK.

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 9 above)
Comment (hein-obox) [reply modules/nested-tabs/assets/scss/frontend.scss:162]: Can't we remove this and move it to line 233 as well?

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 10 above)
Comment (hein-obox) [reply modules/nested-tabs/assets/scss/frontend.scss:194]: Can't we remove this and move it? Don't we have this in line 241 already?

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 14]
```diff
@@ -229,6 +229,21 @@
 	}
 }
 
+// TODO: Remove once PHP global color support is fixed.
+:is( .elementor .elementor-element.elementor-widget-n-tabs > .elementor-widget-container > .e-n-tabs > .e-n-tabs-heading .e-n-tab-title,
+  .elementor .elementor-element.elementor-widget-n-tabs > .elementor-widget-container > .e-n-tabs > .e-n-tabs-content .e-n-tab-title ){
+
```
Comment (hein-obox) [inline modules/nested-tabs/assets/scss/frontend.scss:235]: Can we add the normal styling here as well?

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 4 above)
Comment (hein-obox) [reply modules/nested-tabs/widgets/nested-tabs.php:88]: Can we try it again? I believe it should work.

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 7 above)
Comment (hein-obox) [reply modules/nested-tabs/widgets/nested-tabs.php:429]: We use a css variable here, but for the hover and active state we use a php selector.

Couldn't we use the same approach everywhere?

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 3 above)
Comment (davseve) [reply modules/nested-tabs/widgets/nested-tabs.php]: I added this by mistake, AFAIR it doesn't appear in the original branch

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 7 above)
Comment (davseve) [reply modules/nested-tabs/widgets/nested-tabs.php:429]: I am afraid we can't, if you remember, yesterday, we tried using the PHP approach in normal background color but it didn't work.
The use of the CSS variable is because we don't have a global variable in the light gray color.
This way the tab's normal background color is the BG fallback and if a color is selected that we assign the variable that appears in the PHP

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 14 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:235]: We don't need to. 
This solution solves the malfunction of the PHP global colors, in the normal state we are not using global colors.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js (hunk 12 above)
Comment (davseve) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.js]: Great suggestion.
Done.

File: modules/nested-tabs/assets/scss/frontend.scss [hunk 15]
```diff
@@ -229,6 +229,21 @@
 	}
 }
 
+// TODO: Remove once PHP global color support is fixed.
```
[the remark this answers is not in the document]
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss:232]: Removed

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 4 above)
Comment (davseve) [reply modules/nested-tabs/widgets/nested-tabs.php:88]: We tried together :)
We learned that the specificity sign can't the last element of an `:is` function.
Please resolve.

File: modules/nested-tabs/assets/scss/frontend.scss (hunk 2 above)
Comment (davseve) [reply modules/nested-tabs/assets/scss/frontend.scss]: We have decided to leave it because we are using it to fill the lack of PHP global colors.

**Review by hein-obox — DISMISSED** [review]

**Review by TzviRabinovitch — DISMISSED** [review]

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 16]
```diff
@@ -619,7 +634,7 @@ protected function register_controls() {
 			'global' => [
 				'default' => Global_Typography::TYPOGRAPHY_ACCENT,
 			],
-			'selector' => "{$nested_tabs_heading_selector_class} .e-n-tab-title-text",
+			'selector' => "{$nested_tabs_heading_selector_class} > .e-n-tab-title-text",
```
Comment (hein-obox) [inline modules/nested-tabs/widgets/nested-tabs.php:637]: `'selector' => "{$nested_tabs_heading_selector_class} > .e-n-tab-title > .e-n-tab-title-text",`

**Review by hein-obox — DISMISSED** [review]

**Review by TzviRabinovitch — DISMISSED** [review]

**Review by hein-obox — DISMISSED** [review]

**Review by TzviRabinovitch — APPROVED** [review]
